File: sdkwork-app-sdk-java/src/main/java/com/sdkwork/app/api/PasswordApi.java
package com.sdkwork.app.api;

import com.sdkwork.app.http.HttpClient;
import com.sdkwork.app.model.*;
import java.util.List;
import java.util.Map;

public class PasswordApi {
    private final HttpClient client;
    
    public PasswordApi(HttpClient client) {
        this.client = client;
    }

    /** 修改密码 */
    public PlusApiResultVoid change(PasswordChangeForm body) throws Exception {
        return (PlusApiResultVoid) client.put(ApiPaths.appPath("/user/password"), body);
    }

    /** 重置密码 */
    public PlusApiResultVoid reset(PasswordResetForm body) throws Exception {
        return (PlusApiResultVoid) client.post(ApiPaths.appPath("/auth/password/reset"), body);
    }
}

File: sdkwork-app-sdk-kotlin/src/main/kotlin/com/sdkwork/app/api/PasswordApi.kt
package com.sdkwork.app.api

import com.sdkwork.app.*
import com.sdkwork.app.http.HttpClient

class PasswordApi(private val client: HttpClient) {

    /** 修改密码 */
    suspend fun change(body: PasswordChangeForm): PlusApiResultVoid? {
        return client.put(ApiPaths.appPath("/user/password"), body) as? PlusApiResultVoid
    }

    /** 重置密码 */
    suspend fun reset(body: PasswordResetForm): PlusApiResultVoid? {
        return client.post(ApiPaths.appPath("/auth/password/reset"), body) as? PlusApiResultVoid
    }
}
